Eevee.Sleep.Bot/Handlers/InteractionHandler.cs

```diff
@@ -116,4 +119,25 @@ await interaction
                 throw new ArgumentException($"Unhandled modal ID: {modalId}");
         }
     }
+
+    private static async Task OnButtonClicked(SocketMessageComponent component)
+    {
+        var customId = component.Data.CustomId;
+        if (!customId.StartsWith("role"))
+        {
+            return;
+        }
+
+        await component.RespondAsync("You have been given the role!", ephemeral: true);
+
+        var roleId = ulong.Parse(customId.Replace("role", ""));
+        var user = (SocketGuildUser)component.User;
+
+        await user.RemoveRolesAsync(
```

Comment:
Maybe refactor all these add/remove to a class that is dedicated to handle the role changes (like a bridge of role record controller and the main implementation of Discord interactions)

Reply:
Included here

I've transferred the process to `ButtonClickHandler`, is that what you mean?
[042bf5a](https://github.com/RaenonX-PokemonSleep/eevee-sleep-discord-bot/pull/25/commits/042bf5ae19def3a76b4fa219af2021bdffdd459e)